app/code/Magento/CmsGraphQl/Model/Resolver/Blocks.php

```diff
@@ -78,11 +90,29 @@ private function getBlocksData(array $blockIdentifiers): array
         $blocksData = [];
         try {
             foreach ($blockIdentifiers as $blockIdentifier) {
-                $blocksData[$blockIdentifier] = $this->blockDataProvider->getData($blockIdentifier);
+                $blockData = $this->blockDataProvider->getData($blockIdentifier);
+                if (!empty($blockData)) {
+                    $blocksData[$blockIdentifier] = $blockData;
+                } else {
+                    $this->setErrorMessage(sprintf('The CMS block with the "%s" ID is disabled.', $blockIdentifier));
```

Comment:
`\Magento\Framework\GraphQl\Exception\GraphQlNoSuchEntityException` seems to be suitable here. And please note that it should be translatable (use `__()`)